### PR TITLE
Starts on GroupFinder component

### PR DESCRIPTION
**Pull request by aaronschachter**
### What's this PR do?

This pull request moves the `GroupSelect` component rendered in the `CampaignSignupForm` into a new `GroupFinder` component, in anticipation of adding an additional [UsaStateSelect](https://github.com/DoSomething/phoenix-next/blob/master/resources/assets/components/utilities/UsaStateSelect.js) utility component (refs #2238). 

The `UsaStateSelect` will be used filter the group search by selected state, if the group type's `filterByState` attribute is to `true` ~~(this is blocked from merge until https://github.com/DoSomething/graphql/pull/250 is deployed)~~ UPDATE: It's been deployed

### How should this be reviewed?

:eyes:

### Any background context you want to provide?

No functional changes yet, this is just prep work for adding the state dropdown, and to help keep the PR's reviewable.

### Relevant tickets

References [Pivotal #173231765](https://www.pivotaltracker.com/story/show/173231765).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added screenshots of front-end changes on small, medium, and large screens.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

**Comment by aaronschachter** [comment]
The Cypress tests are failing as the campaign banner displays a GraphQL error: `Cannot query field "filterByState" on type "GroupType"`. Once https://github.com/DoSomething/graphql/pull/250 is merged, I'll update our `schema.json` with the new GraphQL changes which should fix the tests.

**Comment by aaronschachter** [comment]
This is ready for review -- https://github.com/DoSomething/graphql/pull/250 has been deployed, and I've updated this branch's tests and schema.json with the relevant GraphQL updates.

File: resources/assets/components/CampaignBanner/CampaignBanner.js [hunk 1]
```diff
@@ -27,6 +27,10 @@ const CAMPAIGN_BANNER_QUERY = gql`
     campaign(id: $id) {
       id
       groupTypeId
+      groupType {
+        id
+        filterByState
```
Comment (Dinnall) [inline resources/assets/components/CampaignBanner/CampaignBanner.js:32]: Everything looks straightforward, I just had a question about what you mentioned in the PR description. Should this be set to `true` to match the PR description?

Comment (aaronschachter) [reply resources/assets/components/CampaignBanner/CampaignBanner.js:32]: The value of this `filterByState` field will be set to whatever the Group Type's `filter_by_state` boolean field is set to in Rogue admin (added over in https://github.com/DoSomething/rogue/pull/1057) via the create/edit Group Type form.

Comment (Dinnall) [reply resources/assets/components/CampaignBanner/CampaignBanner.js:32]: ah - gotcha. thanks for clarifying

**Pull request merged**